okhttp-client/src/main/scala/org/http4s/client/okhttp/OkHttp.scala

```diff
@@ -0,0 +1,182 @@
+package org.http4s.client.okhttp
+
+import java.io.IOException
+
+import cats.data._
+import cats.effect._
+import cats.implicits._
+import cats.effect.implicits._
+import okhttp3.{
+  Call,
+  Callback,
+  OkHttpClient,
+  Protocol,
+  RequestBody,
+  Headers => OKHeaders,
+  MediaType => OKMediaType,
+  Request => OKRequest,
+  Response => OKResponse
+}
+import okio.BufferedSink
+import org.http4s.{Header, Headers, HttpVersion, Method, Request, Response, Status}
+import org.http4s.client.{Client, DisposableResponse}
+import fs2.Stream._
+import fs2._
+import fs2.io._
+import org.log4s.getLogger
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+
+object OkHttp {
+
+  private val logger = getLogger
+
+  def defaultConfig[F[_]]()(implicit F: Effect[F]): F[OkHttpClient.Builder] = F.delay {
+    new OkHttpClient.Builder()
+      .protocols(
+        List(
+          Protocol.HTTP_2,
+          Protocol.HTTP_1_1
+        ).asJava)
+  }
+
+  /** Create an okhttp client with the default config */
+  def apply[F[_]]()(implicit F: Effect[F], ec: ExecutionContext): F[Client[F]] =
+    apply(defaultConfig[F]())
+
+  /** Create an okhttp client with a supplied config */
+  def apply[F[_]](
+      config: F[OkHttpClient.Builder])(implicit F: Effect[F], ec: ExecutionContext): F[Client[F]] =
+    F.map(config) { c =>
+      val client = c.build()
+      Client(
+        Kleisli { req =>
+          F.async[DisposableResponse[F]] { cb =>
+            client.newCall(toOkHttpRequest(req)).enqueue(handler(cb))
+            ()
+          }
+        },
+        F.delay({
+          try {
```

Comment:
Should we accumulate these errors here? Seems like we're swallowing an awful lot of stuff.

Reply:
They're all being logged, and `OkHttp` eventually shuts down idle resources anyway. See discussion on #1851.  I think we're fine here.